sql/src/main/java/io/crate/protocols/postgres/ConnectionContext.java

```diff
@@ -159,32 +161,44 @@
 
     private static final Logger LOGGER = Loggers.getLogger(ConnectionContext.class);
 
-    final MessageDecoder decoder;
-    final MessageHandler handler;
     private final SQLOperations sqlOperations;
     private final Authentication authService;
 
+    private ChannelPipeline pipeline;
+    private SslHandler sslHandler;
     private int msgLength;
     private byte msgType;
     private SQLOperations.Session session;
     private boolean ignoreTillSync = false;
     private SessionContext sessionContext;
 
     enum State {
-        SSL_NEG,
+        PRE_STARTUP,
         STARTUP_HEADER,
         STARTUP_BODY,
         MSG_HEADER,
         MSG_BODY
     }
 
-    private State state = STARTUP_HEADER;
+    private State state = PRE_STARTUP;
 
-    ConnectionContext(SQLOperations sqlOperations, Authentication authService) {
+    private ConnectionContext(Channel channel, SslHandler sslHandler, SQLOperations sqlOperations, Authentication authService) {
+        this.pipeline = channel.pipeline();
         this.sqlOperations = sqlOperations;
+        this.sslHandler = sslHandler;
         this.authService = authService;
-        decoder = new MessageDecoder();
-        handler = new MessageHandler();
+        // initialize the Netty pipeline with this context's handlers
+        pipeline.addLast("frame-decoder", new MessageDecoder());
+        pipeline.addLast("handler", new MessageHandler());
```

Comment:
What's the advantage of having this registration here instead of in the ChannelInitializer? To introduce the SslHandler this change shouldn't be necessary?